src/java.base/share/classes/jdk/internal/module/SystemModuleFinders.java

```diff
@@ -432,11 +432,27 @@ private ImageLocation findImageLocation(String name) throws IOException {
             }
         }
 
+        /**
+         * Verifies the existence of the given resource, {@code false}
+         * if not found.
+         */
+        private boolean containsLocation(String name) throws IOException {
```

Comment:
Can you rename this to containsImageLocation to keep it consistent with findImageLocation? Alternative rename findImageLocation. Also would be better for the description to be "Returns true if the given resource exists, false if not found".

The changes to the jimage code will take time to review, probably should have 2 reviewers.

Reply:
Done. 

I've looped in @JimLaskey